src/spectral.ts

```diff
@@ -42,7 +43,18 @@ export class Spectral {
     this._resolver = opts && opts.resolver ? opts.resolver : new Resolver();
   }
 
-  public async run(target: IParsedResult | object | string, opts: IRunOpts = {}): Promise<IRuleResult[]> {
+  public async run(
```

Comment:
Whaaaaaat is this. Does TypeScript have overloading? 

I have a funny feeling in my stomach, and I don't think it's just because of the North Sea. Did you consider alternate methods which can be called to return the resolved doc and results?

Reply:
Actually, all of them have some pros and cons. 
For me, the most reasonable one would be to have `getResolved` method, but the gotcha here is that one would need to call `run` beforehand.